data/js/skill/mining.js

```diff
@@ -379,6 +443,35 @@ function onTimer( mChar, timerID )
 						// Remove some ore from the ore resource
 						mResource.oreAmount = mResource.oreAmount - 1;
 
+						var mItem;
+						var totalChance = 0;
+
+						for( mItem = mChar.FirstItem(); !mChar.FinishedItems(); mItem = mChar.NextItem() )
```

Comment:
Don't bother looping through items looking for stone mining bonus if player has not already learned how to mine stone. Move all this inside the "if( mChar.GetTag( "GatheringStone" )" block?